### ISSUE TITLE
Non-string values are being encoded as CData when stringEncodingStrategy = .cdata

### ISSUE THREAD

**Issue by ksoftllc** [MaxDesiatov/XMLCoder#178]
Example:
 
```
import Foundation
import XMLCoder

struct CData: Codable {
    let string: String
    let int: Int
    let bool: Bool
}

let example = CData(string: "string", int: 123, bool: true)
let xmlEncoder = XMLEncoder()
xmlEncoder.stringEncodingStrategy = .cdata
xmlEncoder.outputFormatting = .prettyPrinted
let encoded = xmlEncoder.encode(example)
let encodedAsString = String(data: encoded, encoding: .utf8)
print(encodedAsString!)
```

Produces:
```
<CData>
    <string><![CDATA[string]]></string>
    <int><![CDATA[123]]></int>
    <bool><![CDATA[true]]></bool>
</CData>
```

Expected:
```
<CData>
    <string><![CDATA[string]]></string>
    <int>123</int>
    <bool>true</bool>
</CData>
```

**Comment by MaxDesiatov** [comment]
Thanks for reporting this @ksoftllc, I'll have a look.

**Comment by MaxDesiatov** [comment]
@ksoftllc this is now fixed in `master` and will be released as a part of 0.11.1. Thanks again!

**Comment by ksoftllc** [comment]
Thank you. Nice library.
